### ISSUE TITLE
FON - Ständige Probleme

### ISSUE THREAD

**Issue by goosefraba** [a-sit-plus/at-registrierkassen-mustercode#520]
Hat jemand derzeit ähnliche Probleme mit FON?

- Sporadische "Technische Fehler" als Response auf fast jeden Request
- Im Fall eines technischen Fehlers, nochmal den Request schicken, dann geht er durch
- Mitten unterm Tag einfach mal in den " Wartungsmodus" gehen und alle Requests ablehnen

Wie soll man hier gesetzeskonform arbeiten, wenn die offizielle Stelle nicht in der Lage ist ein stabiles Service zu bieten?!

**Comment by csupnig** [comment]
Das Problem mit den "Technische Fehler"-Responses hatten wir auch. Wir haben hierfür auch noch keine Lösung und schon einige Zertifikate verschwendet. Auch die Kunden, bei denen eine Registrierung der Kasse aufgrund eines solchen "technischen" Fehlers nicht funktioniert hat, sind verunsichert.

**Comment by tragatschnig** [comment]
Ja, wir konnten das auch beobachten.
Da wir bereits damit gerechnet haben, lösen wir das Problem, indem die Anmeldung bei FON komplett asynchron durchläuft. Im Fehlerfall von Seiten von FON wird die Anmeldung zu einem späteren Zeitpunkt wiederholt - welche dann hoffentlich in 7 Tagen vollständig durchläuft, sofern FON up and running ist.
Damit lassen sich zwar auch nicht 100% der Fehlerfälle abdecken, aber es entspannt die Situation deutlich.

**Comment by MartinDomig** [comment]
@tragatschnig wird die Verwendung der Kasse unterbunden, bis die Registrierung bei FON durch ist? Wenn nicht könnte es nämlich sein, dass die Prüfung der Belege, die zwischenzeitlich erstellt wurden, fehl schlägt, hier gibt es noch [offene Fragen](https://github.com/a-sit-plus/at-registrierkassen-mustercode/issues/428#issuecomment-286669492) wie genau die RKSV hier ausgelegt wird.

**Comment by tragatschnig** [comment]
@MartinDomig Nein, die Verwendung der Kasse wird nicht unterbunden, da theoretisch FON auch ein paar Tage down sein könnte (es gab scheinbar in der Vergangenheit Wartungsfenster, welche das gesamte Wochenende gedauert haben). 
Meist ist die gesamte Prozedur aber in wenigen Sekunden abgeschlossen.

Das ist korrekt - die von Ihnen beschriebene Möglichkeit besteht. Wir prüfen mit allen möglichen Mitteln, dass keine fehlerhaften Daten (z.B. nicht mehr gültige UID) vorliegen. 100%ig ausschließen können wir jedoch nicht, dass eine Prüfung fehl schlägt. Die Wahrscheinlichkeit ist aber sehr stark reduziert, dass dies eintritt.
Der noch nicht handlebare Fall ist, dass ein Zertifikat auf eine Steuernummer ausgestellt ist, und mit einem nicht der Steuernummer zugeordnetem FON-Zugang geprüft wird. Dieser Fall lässt sich jedoch technisch nicht vorab prüfen und bedarf organisatorischer Vorbereitung (oder Nachbereitung im Fehlerfall :D ).
